plugin-maven/src/main/java/com/diffplug/spotless/maven/generic/ReplaceRegex.java

```diff
@@ -0,0 +1,28 @@
+package com.diffplug.spotless.maven.generic;
+
+import com.diffplug.spotless.FormatterStep;
+import com.diffplug.spotless.generic.ReplaceRegexStep;
+import com.diffplug.spotless.maven.FormatterStepConfig;
+import com.diffplug.spotless.maven.FormatterStepFactory;
+import org.apache.maven.plugins.annotations.Parameter;
+
+public class ReplaceRegex implements FormatterStepFactory {
+
+	@Parameter
```

Comment:
`@Parameter(required = true)` can probably be used for all fields in this class

Reply:
I cannot reply to your last question below. I don't know why.

The class `Format`  aggregates all generic steps, which can be used to build custom formats, but also for `<java>`, `<scala>`, `<kotlin>` or `<markdown>` if you like. If using a formatter step like eclipse or googleFormat this won't make any sense most of the time. But formatting an markdown file might use some of the generic steps.

Moreover IMHO the maven-plugin should support the same feature set as the gradle-plugin does. And the gradle-plugin supports it this way.

So one could setup the formatting like this
```xml
<configuration>
<!-- global defaults -->
  <encoding>UTF-8</encoding>

  <!-- custom format for properties -->
  <format>
    <inlcudes>
      <include>src/**/resources/**/*.properties</include>
    </includes>
    <endWithNewline />
    <encoding>ISO 8859-1</encoding>
  </format>

  <!-- custom format for java files -->
  <java>
    <endWithNewline />
    <trimTrailingWhitespace />
  </java>
</configuration>
```

Reply:
see above